ktor-features/ktor-server-sessions/jvm/src/io/ktor/sessions/CacheStorage.kt

```diff
@@ -6,13 +6,16 @@ package io.ktor.sessions
 
 import io.ktor.util.*
 import io.ktor.utils.io.*
-import io.ktor.utils.io.jvm.javaio.*
 
 @Suppress("KDocMissingDocumentation")
 @InternalAPI
 public class CacheStorage(public val delegate: SessionStorage, idleTimeout: Long) : SessionStorage {
     private val referenceCache = SoftReferenceCache<String, ByteArray> { id ->
-        delegate.read(id) { input -> input.toInputStream().readBytes() }
+        delegate.read(id) { input ->
+            val result = ByteArray(input.availableForRead)
```

Comment:
So we only read small amount of bytes instead of the full content. Available are bytes available for read at the moment, while the complete size could be unknown